src/browser-realm.ts

```diff
@@ -191,46 +65,26 @@ function removeIframe(iframe: HTMLIFrameElement) {
     }
 }
 
-
 export default function createSecureEnvironment(distortionMap?: Map<RedProxyTarget, RedProxyTarget>, endowments?: object): (sourceText: string) => void {
     const iframe = createDetachableIframe();
-
-    // For Chrome we evaluate the `window` object to kickstart the realm so that
-    // `window` persists when the iframe is removed from the document.
-    const redGlobalThis = (iframe.contentWindow as WindowProxy).window;
-    const { eval: redIndirectEval } = redGlobalThis;
-    redIndirectEval('window');
-    const blueGlobalThis = globalThis;
-
-    removeIframe(iframe);
-
-    const blueRefs = getCachedReferences(blueGlobalThis);
-    const redRefs = getCachedReferences(redGlobalThis);
-
+    const blueWindow = window;
+    const redWindow = (iframe.contentWindow as WindowProxy).window;
+    const endowmentsDescriptors = getFilteredEndowmentDescriptors(endowments || {});
+    const { eval: redIndirectEval } = redWindow;
+    // extra the global references and descriptors before any interference
+    const blueRefs = getCachedReferences(blueWindow);
+    const redRefs = getCachedReferences(redWindow);
+    // creating a new environment
     const env = new SecureEnvironment({
-        blueGlobalThis,
-        redGlobalThis,
+        blueGlobalThis: blueWindow,
+        redGlobalThis: redWindow,
         distortionMap,
     });
-
-    const globalDescriptors = aggregateGlobalDescriptors(
-        redRefs.windowDescriptors,
-        blueRefs.windowDescriptors,
-        endowments && getOwnPropertyDescriptors(endowments)
-    );
-
-    // remapping globals
-    env.remap(redRefs.window, blueRefs.window, globalDescriptors);
-
-    // remapping unforgeable objects
-    env.remap(redRefs.document, blueRefs.document); // no descriptors needed, document.location is unforgeable
-    env.remap(redRefs.EventTargetProto, blueRefs.EventTargetProto, blueRefs.EventTargetProtoDescriptors);
-    env.remap(redRefs.WindowPropertiesProto, blueRefs.WindowPropertiesProto, blueRefs.WindowPropertiesProtoDescriptors);
-    env.remap(redRefs.WindowProto, blueRefs.WindowProto, blueRefs.WindowProtoDescriptors);
-
-    // adjusting proto chains when possible
-    ReflectSetPrototypeOf(redRefs.document, env.getRedValue(blueRefs.DocumentProto));
-
+    linkIntrinsics(env, blueWindow, redWindow);
+    linkUnforgeables(env, blueRefs, redRefs);
+    tameDOM(env, blueRefs, redRefs, endowmentsDescriptors);
+    // once we get the iframe info ready, and all mapped, we can proceed to detach the iframe
+    removeIframe(iframe);
```

Comment:
this is not super relevant, but we are now removing the iframe after all the set up is ready. In master, we remove the iframe as soon as we get the window reference. This seems safer, and more predictable for browsers IMO.

Reply:
Makes sense!

Related but not the same, I've been thinking for local debug purposes to detach the iframe only when `process.env.NODE_ENV !== 'production'`, and also maybe do other safety checks, more logging, etc.

Reply:
Correction: "_...detach the iframe only when `process.env.NODE_ENV === 'production'`..._"

Reply:
yeah, we can do that, what I will do is to make that configurable in this library, because this library doesn't know anything about prod/dev, and locker/sandbox can pass that configuration (keepAlive: true) when needed.